### PR TITLE
Creating Project Cards Dynamically

### PR DESCRIPTION
**Pull request by shahzaman900**
**Creating Project Cards Dynamically**
**In this PR I work on**

1. Recreate card design through JavaScript 
2. Create a popup when clicking on a specific card for a desktop version
3. Create a popup when clicking on a specific card for a desktop version

### CODE REVIEW

**Review by V-Blaze — CHANGES_REQUESTED** [review]
a

File: Projects.js [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+  modal.appendChild(project);
+  setTimeout(() => {
+    const span = document.getElementById('modelcrossIcon');
+    // eslint-disable-next-line func-names
+    span.onclick = function () {
+      modal.style.display = 'none';
+      modal.removeChild(project);
+    };
+  }, 50);
+}
```
Comment (GzimAsani) [inline Projects.js]: - [ ] According to the project requirements you need to implement a validation that the form is to submit only when the email is all in lower case. You have not implemented this, kindly do. Also don't forget to implement the error message
image

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello, @sheezu900                        :wave:   
 Good job so far,  done There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job with the design  +1: 

 ## Required Changes 

- [ ] Check for the comments below :arrow_double_down: 

 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- N/A



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: Projects.js (hunk 1 above)
Comment (shahzaman900) [reply Projects.js]: sorry this PR was for the popup task.

File: Projects.js [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+              src=${card.image}
+              alt="Tonic Project"
+              />
+              <div class="card_detail">
+              <div class="card_title">
+              <h2>${card.Projectname}</h2>
+                <div class="card_history">
+                  <div class="card_history_detail"><p>${card.history.proName}</p></div>
+                <div>
+                   <img src="./images/Counter.svg" alt="dot" />
```
Comment (V-Blaze) [inline Projects.js:93]: - [ ] You have done great so far, :+1: :+1:  but there are some changes you need to make to your UI before moving to the next milestone.
- - As part of the requirement for this milestone you need to replicate the given design as much as you can, currently, in your desktop version of the popup, the technology list and the buttons are aligned to the ring according to the Figma design, but it's not same in your project, I'll suggest you update this. :+1: :smile: 

- - Your Design:
Selection_067

- - Figma design:
Screenshot from 2023-06-21 16-07-00


- - Also according to the Figma design, in the mobile version, there is a Horizontal line between the text and the buttons as shown in the Figma design, I'll suggest you update this in your design too. :+1: :+1: 
- - Figma design:
Selection_068

File: Projects.js [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+    return section.appendChild(project);
+  });
+};
+
+projectMethod();
+
+const modal = document.getElementById('model');
+
+// eslint-disable-next-line no-unused-vars
+function button(id) {
```
Comment (V-Blaze) [inline Projects.js]: - According to Microverse, you are not meant to use any linters disable rule unless it was stated. I'll suggest you remove this and try to find the best way to fix this as linteres are meant to enforce the best practices. :+1: :smile:

File: Projects.js [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+                   <button class="modelbtn" type="button"> See source <img style="height: 24px; width: 24px; margin-left: 6px;" src="./images/see source.svg" alt=""> </button>
+         </div>
+          </div>
+       </div>`;
+
+  modal.appendChild(project);
+  setTimeout(() => {
+    const span = document.getElementById('modelcrossIcon');
+    // eslint-disable-next-line func-names
+    span.onclick = function () {
```
Comment (V-Blaze) [inline Projects.js]: - As i suggested above, According to Microverse, you are not meant to use any linters disable rule unless it was stated. I'll suggest you remove this and try to find the best way to fix this as linters are meant to enforce the best practices. :+1: :smile:

File: index.html [hunk 5]
```diff
@@ -102,190 +102,9 @@ <h1 class="heading">I'm Shah Zaman. Glad to see you!</h1>
...
-						</div>
-					</div>
-					<div class="card_action">
-						<button type="button">See Project</button>
-					</div>
-				</div>
-			</div>
+		<section id="works"></section>
+		<section id="model">
+			<!-- <div id="modelsection" class="modelsection"></div> -->
```
Comment (V-Blaze) [inline index.html]: - [ ] Following the best practices, you are not meant to leave unused code as a comment in your codebase, I'll suggest you remove this. :+1: :wave:

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @sheezu900 , 

## Required changes 🖊️ 
# Almost there 🟡 🟦 🟨 

almost-there


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing the project requirements, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
- Pop opens and closes fine✔
- Work section projects are generated randomly👋 
-  Readme looks professional✔

 ## Required Changes ♻️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: style.css
Comment (Kamba56) [inline style.css:1]: - [ ] As suggested by the previous reviewer, kindly add a line between the list of technologies and buttons as it appears in the figma template. You are expected to stick to design template as much as possible.
### Your design
image
### Figma
image

**Review by Kamba56 — CHANGES_REQUESTED** [review]
Hi @sheezu900, 

Great work on making the changes requested by a previous reviewer 👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- [x] Great job adding the projects as an array in javascript👏👏👏
- [x] Good job implementing the popup window.!👏👏👏

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: Projects.js [hunk 6]
```diff
@@ -176,20 +176,21 @@ function button(id) {
...
+                <span class="small"
                  >${card.tags.javascript}</span
                        >
-               </div>
-                 </div>
-             <div class="card_action">
+              </div>
+            </div>
+            <hr class="hr">
+            <div class="card_action">
```
Comment (usorfaitheloho) [inline Projects.js:193]: As suggested by the previous reviewer, kindly add a horizontal line and style it as shown in the figma design given , it's always good that your design looks close to the template given to show good attention to detail 😄 

Figma design :
image

Your design :
image

**Review by usorfaitheloho — CHANGES_REQUESTED** [review]
> Hi @sheezu900  ,
> 
> Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!

  [ ](https://camo.githubusercontent.com/679757134a8fee76cfe1941f9f3ecb0e0cf56dd02da4498268ce31f0dacdfca3/68747470733a2f2f6d65646961302e67697068792e636f6d2f6d656469612f556461535559436c723932636a57706556782f67697068792e676966)
> 
> ## Highlights
> * Good javascript code quality✔️
> * Descriptive pull request ✔️
> * Used GitHub flow ✔️
> 
> ## Required Changes ♻️
> _Check the comments under the review._
> 
> ## Optional suggestions
> _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
> 
> Cheers and Happy coding!👏👏👏
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
> 
> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
> 
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by LionRouge1 — APPROVED** [review]
Hi @sheezu900, 

# STATUS: APPROVED! 🥇 💯 
Great work on making the changes 👏 
You've done well implementing the requested changes

done

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**